x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/EsqlFunctionRegistry.java

```diff
@@ -169,6 +170,8 @@ private FunctionDefinition[][] functions() {
                 def(DateParse.class, DateParse::new, "date_parse"),
                 def(DateTrunc.class, DateTrunc::new, "date_trunc"),
                 def(Now.class, Now::new, "now") },
+            // spatial
+            new FunctionDefinition[] { def(SpatialCentroid.class, SpatialCentroid::new, "st_centroid") },
```

Comment:
I choose this location as directly after the `// date` section, since that is the only other cases of a type-specific set of additional functions. Seems like the right location to me. If we move it to the end, should we move `// date` to the end too?